src/ORM/Association/BelongsToMany.php

```diff
@@ -228,8 +228,32 @@ public function getForeignKey()
 
     /**
      * Sets the sort order in which target records should be returned.
+     *
+     * @param mixed $sort A find() compatible order clause
+     * @return $this
+     */
+    public function setSort($sort)
+    {
+        $this->_sort = $sort;
+
+        return $this;
+    }
+
+    /**
+     * Gets the sort order in which target records should be returned.
+     *
+     * @return mixed
+     */
+    public function getSort()
+    {
+        return $this->_sort;
+    }
+
+    /**
+     * Sets the sort order in which target records should be returned.
      * If no arguments are passed the currently configured value is returned
      *
+     * @deprecated 3.4.1 Use setSort()/getSort() instead.
```

Comment:
I have already done this change in 3.5. I think we should avoid deprecating some methods in the bugfix release.
https://github.com/cakephp/cakephp/pull/10242